lib/temporal/worker/activity.rb

```diff
@@ -0,0 +1,80 @@
+require 'temporal/worker/sync_worker'
+require 'temporal/worker/activity_task_processor'
+
+module Temporal
+  class Worker
+    class Activity
+      def initialize(core_worker, converter, executor)
+        @running = false
+        @worker = SyncWorker.new(core_worker)
+        @converter = converter
+        @executor = executor
+      end
+
+      def run(reactor)
+        @running = true
+
+        while running?
+          activity_task = worker.poll_activity_task
+          reactor.async do
+            if activity_task.start
+              handle_start_activity(activity_task.task_token, activity_task.start)
+            elsif activity_task.cancel
+              handle_cancel_activity(activity_task.task_token, activity_task.cancel)
+            end
+          end
+        end
+      end
+
+      def shutdown
+        @running = false
+        executor.shutdown
+        # TODO: core worker shutdown implementation pending
+      end
+
+      private
+
+      attr_reader :worker, :converter, :executor
+
+      def running?
+        @running
+      end
+
+      def process_activity_task(task)
+        queue = Queue.new
+
+        executor.schedule do
+          result = ActivityTaskProcessor.new(task).process
+          queue << result
+        rescue StandardError => e
+          queue << e
+        end
+
+        queue.pop
```

Comment:
IIUC, this blocks the fiber and not the thread since it is run in a `reactor.async` block, right?

Reply:
Nevermind me, I should have read the discussion above.